api/tests/core/users/test_api.py

```diff
@@ -292,6 +293,38 @@ def test_suspend_pro_with_other_offerer_users(self):
 
         _assert_user_suspension_history(pro, SuspensionEventType.SUSPENDED, reason, actor)
 
+    @pytest.mark.usefixtures("db_session")
+    class PasswordChangeTest:
+        @pytest.mark.parametrize("ff_activation", [True, False])
+        def test_with_feature_flag(self, ff_activation):
+            """
+            Test that an account suspended upon user request keeps the same
+            password when the feature flag is activated.
+            """
+            user = users_factories.BeneficiaryGrant18Factory()
+            reason = users_constants.SuspensionReason.UPON_USER_REQUEST
+            expected_password_hash = user.password
+
+            with override_features(ALLOW_ACCOUNT_REACTIVATION=ff_activation):
+                users_api.suspend_account(user, reason, user)
+
+                user = User.query.get(user.id)
+                assert user.password == expected_password_hash or not ff_activation
```

Comment:
Cet assert ne vérifie pas que `user.password != expected_password`_hash si le FF est désactivé.
Peut-être ajouter aussi un `assert not user.isActive`, puisque c'est quand même le résultat de l'action.